test/jdk/java/net/httpclient/ExpectContinueTest.java

```diff
@@ -0,0 +1,270 @@
+            try (InputStream is = exchange.getRequestBody();
+                 OutputStream os = exchange.getResponseBody()) {
+                byte[] bytes = "417 Expectation Failed".getBytes();
+                exchange.sendResponseHeaders(417, bytes.length);
+                os.write(bytes);
+            }
+        }
+    }
+
+    static class Http1HangServer extends Thread {
+
+        final ServerSocket ss;
+        final InetAddress ia;
+        final int port;
+        volatile boolean closed = false;
+
+        Http1HangServer(InetSocketAddress sa) throws IOException {
+            ss = ServerSocketFactory.getDefault()
+                    .createServerSocket(sa.getPort(), -1, sa.getAddress());
+            ia = ss.getInetAddress();
+            port = ss.getLocalPort();
+        }
+
+        @Override
+        public void run() {
+            byte[] bytes = "417 Expectation Failed".getBytes();
+
+            while (!closed) {
+                try {
+                    // Not using try with resources here as we expect the client to close resources when
+                    // 417 is received
+                    Socket client = ss.accept();
+                    InputStream is = client.getInputStream();
+                    OutputStream os = client.getOutputStream();
+
+                    BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+                    Writer w = new OutputStreamWriter(os, UTF_8);
+                    PrintWriter pw = new PrintWriter(w);
+
+                    StringBuilder response = new StringBuilder();
+                    String line = null;
+                    StringBuilder reqBuilder = new StringBuilder();
+                    while (!(line = reader.readLine()).isEmpty()) {
+                        reqBuilder.append(line + "\r\n");
+                    }
+                    String req = reqBuilder.toString();
+                    System.err.println("Http1HangServer received: " + req);
+                    StringTokenizer tokenizer = new StringTokenizer(req);
+                    String method = tokenizer.nextToken();
+                    String path = tokenizer.nextToken();
+                    String version = tokenizer.nextToken();
+
+                    boolean validRequest = method.equals("POST") && path.equals("/http1/hang")
+                                        && version.equals("HTTP/1.1");
+                    // If correct request, send 417 reply. Otherwise wait for correct one
+                    if (validRequest) {
+                        closed = true;
+                        response.append("HTTP/1.1 417 Expectation Failed\r\n")
+                                .append("Content-Length: ")
+                                .append(0)
+                                .append("\r\n\r\n");
+                        pw.print(response);
+                        pw.flush();
+
+                        os.write(bytes);
+                        os.flush();
+                    }
```

Comment:
maybe you should close the client socket if validRequest == false. You should also implement a close() method to close the server socket and the client socket when the test terminates. Also the test seems to be missing a teardown() method to properly close and stop all the servers.